### DISCUSSION TITLE
Supabase Admin Dashboard in App - RLS

### DISCUSSION THREAD

**Discussion by bennik88** [supabase/supabase#8813] [General]
I worked myself through all your documentation and RLS Videos on Youtube on your Channel and searched a lot over google but I'm absolutely finding no content on how to work with different roles like "admin", "superadmin", "client" or whatever custom roles I want. I'm coming from Backends like Django, Firebase and MongoDB and now read my way through all your Triggers, Serverless functions, and multiple SQL Curses but can't get it to work like Firebase with Custom Claims or Django Role Based Authentication. You're providing an abstract UI for Users who don't want to worry about always diving into PosergreSQL but I'm not able to figure out how to deal with such a basic use case when it comes to authentication. I've also seen your SlackClone but I don't want to work with email address RLS I simply want a role like "superadmin" (so I guess no permissions only roles) which is stored at the user or a related table who is then able to access (create, update, delete) all resources and also set roles for authenticated users. How would I achieve this? I also didn't find any information on how to change the user.role() attribute or extend it. Did I miss something in the docs as this is such a basic usecase? Restricting access for "signed in / authenticated" users with RLS is clear for me.
By inspecting the `auth.user` I found that it includes columns for `role` and `is_super_admin`. How can I set them and use it as an administrative way to delete or deactivate auth.users or let's say blog posts from my app. So I later can check these conditions with RLS like `auth.user.role = "superadmin"`.

**Comment by GaryAustin1** [comment]
This shows one way to implement custom claims in the jwt using the raw_app_meta_data field.  https://github.com/supabase-community/supabase-custom-claims

**Comment by bennik88** [comment]
Thank you @GaryAustin1 I looked at your custom claims link and it is super useful to know. That should work for my use case for now. Is there a reason why this resource isn't in the documentation yet?
I hope there will be a better more advanced documentation to RLS than the Slack Clone in the future. Therefore I would switch to this advanced pattern of RLS. Would be super great to update this thread as soon as this happens. Again thanks.

**Reply by GaryAustin1** [reply]
I don't know how Supabase decides what is in supabase-community and more official.  If you have documentation suggestions I would create a documentation issue, or raise an idea in discussions here.
